File: ViagemMasterData/Models/Vehicles.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DDDNetCore.Models
{
    public class Vehicles
    {
        //matricula, VIN, pelo seu tipo e data de entrada ao serviço
        public string Code { get; set; }
        public string LicencePlate { get; set; }
        public string Vin { get; set; }
        public string VehicleType { get; set; }
        public DateTime StartDate { get; set; }
    }
}

File: viajantes2-spa/src/app/ViagemMasterData/models/vehicles.ts
import { VehicleType } from "src/app/RedeMasterData/models/vehicle-type";

export class Vehicle {
    id: string;
    licencePlate: string;
    vin: string;
    vehicleTypeId: string;
    vehicleType: VehicleType | undefined;
    startDate: string;

    constructor(){
        this.vehicleType = new VehicleType();
    }
}